`master` branch fails to compile after multi-threading changes

### Godot version

4.0 (809a98216267f3066b9fec2f02b2042bdc9d3e0d)

### System information

Windows 10, MSVC 14.2 (VS 2019)

### Issue description

Trying to compile the engine fails with

```
scene/2d/node_2d.h(120): error C2280: 'Node::MTFlag::MTFlag(void)': attempting to reference a deleted function
scene/main/node.h(57): note: compiler has generated 'Node::MTFlag::MTFlag' here
scene/main/node.h(57): note: 'Node::MTFlag::MTFlag(void)': function was implicitly deleted because 'Node::MTFlag' has a variant data 
member 'Node::MTFlag::mt' with a non-trivial default constructor
scene/main/node.h(55): note: see declaration of 'Node::MTFlag::mt'
```

Scons is run as

```
scons platform=windows -j6 target=editor dev_build=yes
```

### Steps to reproduce

Try to compile the engine from the current `master` with MSVC 14.2 and the following command (probably irrelevant):

```
scons platform=windows -j6 target=editor dev_build=yes
```

### Minimal reproduction project

N/A

--- a/scene/main/node.h
+++ b/scene/main/node.h
@@ -52,13 +52,17 @@ protected:
 	// During group processing, these are thread-safe.
 	// Outside group processing, these avoid the cost of sync by working as plain primitive types.
 	union MTFlag {
-		SafeFlag mt{};
+		SafeFlag mt;
 		bool st;
+		MTFlag() :
+				mt{} {}
 	};
 	template <class T>
 	union MTNumeric {
-		SafeNumeric<T> mt{};
+		SafeNumeric<T> mt;
 		T st;
+		MTNumeric() :
+				mt{} {}
 	};
 
 public:

